src/main/app.js

```diff
@@ -136,3 +138,26 @@ app.on('activate', () => {
     windowManager.buildWindowOfType('main');
   }
 });
+
+// Prevent quitting if a file is being saved
+// to prevent data loss
+ipc.on('workspace-save-started', () => {
+  isSaving = true;
+});
+
+ipc.on('workspace-save-finished', () => {
+  isSaving = false;
+});
+
+app.on('before-quit', e => {
```

Comment:
I sent a message about this event and cb. If it blocks the user from doing other save-inducing actions, that would be best. Could you confirm the behaviour here? 😊

Reply:
Yeah, the UI disables everything when this happens. Only thing user can do is wait 😂